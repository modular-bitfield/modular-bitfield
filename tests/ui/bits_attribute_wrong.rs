use modular_bitfield::prelude::*;

#[derive_const(Specifier)]
#[derive( Debug)]
pub enum TriggerMode {
    Edge = 0,
    Level = 1,
}

#[bitfield]
pub struct MismatchedTypes {
    #[bits = 9]
    trigger_mode: TriggerMode,
    reserved: B7,
}

const NOT_A_LITERAL: u32 = 9;

#[bitfield]
pub struct InvalidValueType {
    #[bits = NOT_A_LITERAL]
    trigger_mode: TriggerMode,
    reserved: B7,
}

#[bitfield]
pub struct DuplicateAttribute {
    #[bits = 1]
    #[bits = 1]
    trigger_mode: TriggerMode,
    reserved: B7,
}

#[bitfield]
pub struct NotANameValue {
    #[bits(1)]
    trigger_mode: TriggerMode,
    reserved: B7,
}

fn main() {}
