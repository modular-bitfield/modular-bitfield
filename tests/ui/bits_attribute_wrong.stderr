error: encountered invalid value type for #[bits = N]
  --> tests/ui/bits_attribute_wrong.rs:20:14
   |
20 |     #[bits = NOT_A_LITERAL]
   |              ^^^^^^^^^^^^^

error: encountered duplicate `#[bits = ...]` attribute for field
  --> tests/ui/bits_attribute_wrong.rs:28:7
   |
28 |     #[bits = 1]
   |       ^^^^^^^^

error: duplicate `#[bits = ...]` here
  --> tests/ui/bits_attribute_wrong.rs:27:7
   |
27 |     #[bits = 1]
   |       ^^^^^^^^

error: expected `=`
  --> tests/ui/bits_attribute_wrong.rs:35:11
   |
35 |     #[bits(1)]
   |           ^

error[E0308]: mismatched types
  --> tests/ui/bits_attribute_wrong.rs:11:7
   |
11 |     #[bits = 9]
   |       ^^^^^^^^ expected an array with a size of 9, found one with a size of 1
